nep-17.mediawiki

```diff
@@ -0,0 +1,182 @@
+
+==Specification==
+
+In the method definitions below, we provide both the definitions of the functions as they are defined in the contract as well as the invoke parameters.
+
+===Methods===
+
+====symbol====
+
+<pre>
+{
+  "name": "symbol",
+  "parameters": [],
+  "returntype": "String"
+}
+</pre>
+
+Returns a short string representing symbol of the token managed in this contract. e.g. <code>"MYT"</code>. This string MUST be valid ASCII, MUST NOT contain whitespace or control characters, SHOULD be limited to uppercase Latin alphabet (i.e. the 26 letters used in English) and SHOULD be short (3-8 characters is recommended).
+
+This method MUST always return the same value every time it is invoked.
+
+====decimals====
+
+<pre>
+{
+  "name": "decimals",
+  "parameters": [],
+  "returntype": "Integer"
+}
+</pre>
+
+Returns the number of decimals used by the token - e.g. <code>8</code>, means to divide the token amount by <code>100,000,000</code> to get its user representation.
+
+This method MUST always return the same value every time it is invoked.
+
+====totalSupply====
+
+<pre>
+{
+  "name": "totalSupply",
+  "parameters": [],
+  "returntype": "Integer"
+}
+</pre>
+
+Returns the total token supply deployed in the system.
+
+====balanceOf====
+
+<pre>
+{
+  "name": "balanceOf",
+  "parameters": [
+    {
+      "name": "account",
+      "type": "Hash160"
+    }
+  ],
+  "returntype": "Integer"
+}
+</pre>
+
+Returns the token balance of the <code>account</code>.
+
+The parameter <code>account</code> MUST be a 20-byte address. If not, this method SHOULD <code>throw</code> an exception.
+
+If the <code>account</code> is an unused address, this method MUST return <code>0</code>.
+
+====transfer====
+
+<pre>
+{
+  "name": "transfer",
+  "parameters": [
+    {
+      "name": "from",
+      "type": "Hash160"
+    },
+    {
+      "name": "to",
+      "type": "Hash160"
+    },
+    {
+      "name": "amount",
+      "type": "Integer"
+    },
+    {
+      "name": "data",
+      "type": "Any"
+    }
+  ],
+  "returntype": "Boolean"
+}
+</pre>
+
+Transfers an <code>amount</code> of tokens from the <code>from</code> account to the <code>to</code> account.
+
+The parameters <code>from</code> and <code>to</code> MUST be 20-byte addresses. If not, this method SHOULD <code>throw</code> an exception.
+
+The parameter <code>amount</code> MUST be greater than or equal to <code>0</code>. If not, this method SHOULD <code>throw</code> an exception.
```

Comment:
Info about parameter data is missing

Reply:
> If the receiver is a deployed contract, the function MUST call onPayment method on receiver contract with the data parameter from transfer AFTER firing the Transfer event. If the receiver doesn't want to receive this transfer it MUST call ABORT.

Reply:
I mean, whether we need to add a brief description like why we need this parameter, what's the size limitation, etc.